### PR TITLE
build(refs T19759): use vueuse/useSortable instead of vuedraggable [vue 3 upgrade] #682

### PR DESCRIPTION
**Pull request by salisdemos**
**Ticket:** https://yaits.demos-deutschland.de/T35522

- since there is no well maintained version of `vuedraggable` that works with vue 3, we are switching to `vueuse/useSortable` (see https://vueuse.org/integrations/useSortable/)
- event parameters changed, so event handling is adjusted accordingly

https://github.com/demos-europe/demosplan-ui/pull/682/files Closed when I deleted i_vue_integration, I didn'T know that this would happen, so I hat to recreate the PR

### CODE REVIEW

**Review by spiess-demos — CHANGES_REQUESTED** [review]

File: src/components/DpDataTable/DpTableHeader.vue [hunk 1]
```diff
@@ -159,6 +158,12 @@ export default {
     }
   },
 
+  computed: {
+    DpResizableColumn () {
```
[the remark this answers is not in the document]
Comment (salisdemos) [reply src/components/DpDataTable/DpTableHeader.vue]: Its necessary to be loadable via `<component :is...>`. But there is another anotation wich may be more convenient. Is it better like so?
eca81abc8dd999fd82d1a5e5e9821e0f0bad1483

File: src/components/DpDataTable/DpDataTable.vue [hunk 2]
```diff
@@ -102,16 +102,17 @@
...
         :content-data="items"
         handle="c-data-table__drag-handle"
         ghostClass="sortable-ghost"
         chosenClass="sortable-chosen"
-        @change="(e) => $emit('changed-order', e)">
-        <template v-for="(item, idx) in items">
+        @end="(event, item) => $emit('changed-order', event, item)">
+        <template
+          v-for="(item, idx) in items"
+          :key="item.id">
```
[the remark this answers is not in the document]
Comment (salisdemos) [reply src/components/DpDataTable/DpDataTable.vue]: Thanks for spotting. Changed in eca81abc8dd999fd82d1a5e5e9821e0f0bad1483

File: src/components/DpDraggable/DpDraggable.vue [hunk 3]
```diff
@@ -1,138 +1,141 @@
...
-     * Callback to be executed on move event of draggable.
-     * It can be used to cancel drag by returning false.
-     */
-    opts: {
-      type: Object,
-      required: false,
-      default: () => ({})
-    }
-  },
+useSortable(
```
[the remark this answers is not in the document]
Comment (salisdemos) [reply src/components/DpDraggable/DpDraggable.vue]: Thanks for the hint: 43d9b62d9ea00f8d1ed50b1b641524c0a9682b85

File: package.json [hunk 4]
```diff
@@ -8,52 +8,53 @@
...
+    "@vueuse/integrations": "^11.1.0",
     "a11y-datepicker": " ^0.9.0",
     "dayjs": "^1.11.5",
     "dompurify": "^3.0.0",
     "fscreen": "^1.2.0",
     "ismobilejs": "^1.1.1",
     "lscache": "^1.3.2",
     "plyr": "^3.7.2",
-    "prosemirror-model": "^1.21.1",
+    "prosemirror-model": "1.22.3",
```
Comment (spiess-demos) [inline package.json:48]: As a first step to clean up the versioning conflict mess, i'd suggest to get rid of direct access to prosemirror  by [changing this line](https://github.com/demos-europe/demosplan-ui/blob/e624fc6fc892dfe366d8b275bb9bc19a017f68fb/src/components/DpEditor/libs/editorInsertAtCursorPos.js#L1) to 

```
import { DOMParser } from '@tiptap/pm/model'
```

**Review by spiess-demos — CHANGES_REQUESTED** [review]

**Review by spiess-demos — APPROVED** [review]
lgtm (code review).

**Pull request merged**